api/campaigns/class-maybe-show-campaign.php

```diff
@@ -300,21 +306,107 @@ public function should_popup_be_shown( $client_id, $popup, $settings, $referer_u
 		}
 
 		// Handle frequency.
-		$frequency = $popup->f;
-		if ( 'once' === $frequency && count( $seen_events ) >= 1 ) {
+		$frequency         = $popup->f;
+		$frequency_max     = (int) $popup->fm;
+		$frequency_start   = (int) $popup->fs;
+		$frequency_between = (int) $popup->fb;
+		$frequency_reset   = $popup->ft;
+
+		// Override individual settings if a frequency preset is selected.
+		if ( 'once' === $frequency ) {
+			$frequency_max     = 1;
+			$frequency_start   = 0;
+			$frequency_between = 0;
+			$frequency_reset   = 'month';
+		}
+		if ( 'daily' === $frequency ) {
+			$frequency_max     = 1;
+			$frequency_start   = 0;
+			$frequency_between = 0;
+			$frequency_reset   = 'day';
+		}
+		if ( 'always' === $frequency ) {
+			$frequency_max     = 0;
+			$frequency_start   = 0;
+			$frequency_between = 0;
+			$frequency_reset   = 'month';
+		}
+		if ( 'preset_1' === $frequency ) {
+			$frequency_max     = 5;
+			$frequency_start   = 3;
+			$frequency_between = 3;
+			$frequency_reset   = 'month';
+		}
+
+		if ( false === $now ) {
+			$now = time();
+		}
+
+		$reader      = $this->get_reader( $client_id );
+		$seen_events = $this->get_reader_events( $client_id, 'prompt_seen', $popup->id );
+		$total_views = 0;
+
+		// Tally up pageviews of any post type.
+		if ( isset( $reader['reader_data']['views'] ) ) {
+			foreach ( $reader['reader_data']['views'] as $post_type => $views ) {
+				$total_views += (int) $views;
+			}
+		}
+
+		// Guard against invalid or missing reset period values.
+		if ( ! in_array( $frequency_reset, [ 'month', 'week', 'day' ], true ) ) {
+			$frequency_reset = 'month';
+		}
+
+		// Filter seen events for the relevant period.
+		$seen_events = array_filter(
+			$seen_events,
+			function( $event ) use ( $frequency_reset, $now ) {
+				$seen = strtotime( $event['date_created'] );
+				return $seen >= strtotime( '-1 ' . $frequency_reset, $now );
+			}
+		);
+
+		// If not displaying every pageview.
+		if ( 0 < $frequency_between ) {
+			$views_after_start = max( 0, $total_views - ( $frequency_start + 1 ) );
```

Comment:
When setting "Start after pageview" to `1` I'd expect the prompt to be shown _on the second pageview_. Currently it will be shown on the first one.

Reply:
With a "Start after pageview" setting of `1`, the desired behavior should be:

1. First pageview of a session: no prompt
2. Second pageview of the session: prompt

That's what I'm seeing locally and on a new test site. Can you describe steps to replicate other behavior?

Reply:
Sorry, my bad! I was experimenting with the settings end confused "Start after pageview" with "Pageviews between displays".